### ISSUE TITLE
Extending adapters or serializers requires classic decorator

### ISSUE THREAD

**Issue by lowski** [emberjs/ember-classic-decorator#6]
Hi!
I have a question about the `classic` decorator. Documentation says it needs to be used only for the classes that inherits **directly** from `EmberObject`. However when I try to extends from `RESTAdapter` or `RESTSerializer` like in the example below, it raises an error: `Uncaught (in promise) Error: You defined the class ApplicationEmberObject that extends from Ember.Object using native class syntax, but you didn't mark it with the @classic decorator. All user classes that extend from this class must be marked as @classic, since they use classic features. If you want to remove the @classic decorator, see the guides on how to convert this class into the Octane version: [TODO_GUIDE_LINK]`.

```javascript
import classic from 'ember-classic-decorator';
import RESTSerializer from 'ember-data/serializers/rest';

export default class ApplicationEmberObject extends RESTSerializer {
  serializeIntoHash(data, _type, record, options) {
    Object.assign(data, this.serialize(record, options));
  }
}
```

Adding `@classic` on top of the `export default class ApplicationEmberObject extends RESTSerializer` removes the error. Is this a bug?

Btw. I think it would be good to link `TODO_GUIDE_LINK`.

**Comment by pzuraq** [comment]
I believe this is a bug, these are classes that should likely not throw an error if they don't use classic methods

**Comment by basz** [comment]
I seem to have such message too except they are I have no idea from which class it comes as the class seems to be anonymous... `You defined the class _default with a 'constructor' function, but one of its ancestors, DebugFrameworkObject, uses the 'init' method.`.

Is there a way to know which class this is coming from? The stacktrace doesn't help...

**Comment by pzuraq** [comment]
I believe both of these issues should be solved in v1.0.5. I added the Data primitives to the list of classes that can be extended safely, and `DebugFrameworkObject` is a class that was coming from Ember itself, I think for `ObjectProxy`. Also re: the `TODO_GUIDE_LINK`, I removed it, since we don't actually have stable links yet unfortunately. Will add it back in the future.

Let me know if you have any other issues or feedback!
